src/nunit.analyzers/ClassicModelAssertUsage/ClassicModelAssertUsageAnalyzer.cs

```diff
@@ -232,21 +233,58 @@ public sealed class ClassicModelAssertUsageAnalyzer : BaseAssertionAnalyzer
               { NameOfAssertIsNotInstanceOf, isNotInstanceOfDescriptor },
           }.ToImmutableDictionary();
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ClassicModelAssertUsageAnalyzer.name.Values.ToImmutableArray();
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ClassicModelAssertUsageAnalyzer.NameToDescriptor.Values.ToImmutableArray();
 
         protected override void AnalyzeAssertInvocation(SyntaxNodeAnalysisContext context,
             InvocationExpressionSyntax assertExpression, IMethodSymbol methodSymbol)
         {
-            if (ClassicModelAssertUsageAnalyzer.name.ContainsKey(methodSymbol.Name))
+            if (ClassicModelAssertUsageAnalyzer.NameToDescriptor.TryGetValue(methodSymbol.Name, out DiagnosticDescriptor? descriptor))
             {
+                string GetImplicitTypeConversion(ExpressionSyntax? expression)
+                {
+                    if (expression == null)
+                    {
+                        return string.Empty;
+                    }
+
+                    var typeInfo = context.SemanticModel.GetTypeInfo(expression, context.CancellationToken);
+
+                    var convertedType = typeInfo.ConvertedType;
+
+                    if (convertedType == typeInfo.Type)
+                    {
+                        return string.Empty;
+                    }
+
+                    if (convertedType.SpecialType == SpecialType.System_Object ||
+                        convertedType.SpecialType == SpecialType.System_Collections_IEnumerable)
+                    {
+                        return string.Empty;
+                    }
+                    
+                    string implicitType = convertedType.ToString();
+
+                    if (implicitType == "System.Collections.ICollection")
+                    {
+                        return string.Empty;
+                    }
+
+                    return implicitType;
+                }
+
+                var arguments = assertExpression.ArgumentList.Arguments;
+                string firstArgumentImplicitTypeConversion = GetImplicitTypeConversion(arguments[0].Expression);
```

Comment:
I think it should go to CodeFix itself, analyzer doesn't care about this information.

Reply:
Agreed. However the CodeFix has no access to the semantic model to access the necessary type information. Or do you know another way?

Reply:
I suppose you can use `context.Document.GetSemanticModelAsync()`

Reply:
Thanks. Code moved from Analyzer to CoeFix.